frontend/express/app.js

```diff
@@ -1431,6 +1434,75 @@ app.post(countlyConfig.path+'/graphnotes/delete', function (req, res, next) {
     }
 });
 
+app.get(countlyConfig.path+'/render', function(req, res){
+    if (!req.session.uid) {
+        return res.redirect(countlyConfig.path+'/login');
+    }
+    
+    var options = {};
+    var view = req.query.view || "";
+    var route = req.query.route || "";
+    var id = req.query.id || "";
+
+    options.view = view + "#" + route;
+    options.id = id ? "#" + id : "";
+    options.savePath = path.resolve(__dirname, "./public/images/screenshots/screenshot_" + Date.now() + ".png");
+
+    ip.getHost(function(err, host){
+        if(err){
+            console.log(err);
+            return res.send(false);
+        }
+
+        options.host = host;
+
+        authorize.save({db:countlyDb, multi:false, owner:req.session.uid, purpose:"LoginAuthToken", callback:function(err, token){
+            if(err){
+                console.log(err); 
+                return res.send(false);
+            }
+    
+            options.token = token;
+            render.renderView(options, function(err, image){
+                if(err){
+                    return res.send(false);
+                }
+
+                return res.send(true);
+            });
+        }});
+    });
+});
+
+app.get(countlyConfig.path+'/login/token/:token', function(req, res){
+    var token = req.params.token;
+    var pathUrl = req.url.replace(countlyConfig.path, "");
+    var urlParts = url.parse(pathUrl, true);
+    var fullPath = urlParts.pathname;
+
+    authorize.verify_return({db:countlyDb, token:token, req_path:fullPath, callback:function(valid){
+        if(!valid){
+            plugins.callMethod("tokenLoginFailed", {req:req, res:res, data: {token: token}});
+            return res.redirect(countlyConfig.path+'/login?message=login.result');
+        }
+
+        countlyDb.collection('members').findOne({"_id":countlyDb.ObjectID(valid)}, function (err, member) {
+            if(err || !member){
+                plugins.callMethod("tokenLoginFailed", {req:req, res:res, data: {token: token, token_owner: valid}});
+                return res.redirect(countlyConfig.path+'/login?message=login.result');
+            }
+
+            req.session.uid = member["_id"];
```

Comment:
Also due to it being authentication, we should regenerate sesison id to prevent session hijacking, like this:
https://github.com/Countly/countly-server/blob/master/frontend/express/app.js#L968